### PR TITLE
Feature/translations

### PR DESCRIPTION
**Pull request by MAKARD**

### CODE REVIEW

File: src/components/Layout/Partials/Header.tsx [hunk 1]
```diff
@@ -1,21 +1,45 @@
...
 export class Header extends React.Component<undefined, undefined> {
+    public static readonly contextTypes = LayoutContextTypes;
+
+    public context: LayoutContext;
 
     public render(): JSX.Element {
         return (
             <header className="header">
-                <Link to="/" className="logo">
+                <a href="#" className="logo">
```
Comment (Horat1us) [inline src/components/Layout/Partials/Header.tsx]: Why hash?

File: src/components/ContactPage/ContactPage.tsx [hunk 2]
```diff
@@ -5,25 +5,28 @@ import {Config} from "../../data/Config";
...
+    public static readonly contextTypes = LayoutContextTypes;
+    public context: LayoutContext;
 
     public state: ContactPageState = {
         isModalOpen: false,
     };
 
-    public shouldComponentUpdate(nextProps: undefined, nextState: ContactPageState): boolean {
-        return this.state.isModalOpen !== nextState.isModalOpen;
+    public shouldComponentUpdate(nextProps, nextState: ContactPageState, nextContext: LayoutContext): boolean {
```
Comment (Horat1us) [inline src/components/ContactPage/ContactPage.tsx]: It seems `shouldComponentUpdate` does not receive `nextContext` in React16 https://reactjs.org/docs/react-component.html#shouldcomponentupdate

File: src/index.tsx [hunk 3]
```diff
@@ -1,18 +1,31 @@
 import "../styles/main.scss";
+import "bootstrap";
+
+// tslint:disable:no-submodule-imports
+import "translations/en";
```
Comment (Horat1us) [inline src/index.tsx]: use `./transactions/en` instead. do not disable tslint here

File: src/index.tsx [hunk 4]
```diff
@@ -1,18 +1,31 @@
...
 import {PreLoader} from "./components/PreLoader";
+import {Languages} from "./data/Languages";
 
 const container = document.getElementById("content-overlay");
 
 if (!container) {
     // tslint:disable-next-line
     console.error("Can not found element for mounting Layout!");
 } else {
+    if (!localStorage.getItem("app.language")) {
```
Comment (Horat1us) [inline src/index.tsx]: This code shouldn't be here.

File: tests/unit/form-specs.tsx [hunk 5]
```diff
@@ -30,4 +27,21 @@ describe("<Form/>", () => {
...
         expect(isDefaultPrevented).to.be.true;
     });
+
+    it("Should be error on submit when fields filled wrong", async () => {
+        const model = instantiateContactFormModel();
+        // temporary coverage improve
+        model.groups();
+        model.to = model.from = undefined;
+
+        expect((await model.validate()).length).to.equal(model.attributes().length);
```
Comment (Horat1us) [inline tests/unit/form-specs.tsx]: Not real asserts

File: src/components/ContactPage/ContactPage.tsx (hunk 2 above)
Comment (MAKARD) [reply src/components/ContactPage/ContactPage.tsx]: It seems that its big secret if docs does not contain this feature
image

**Pull request merged**